urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/httpd/resource/FileSystemResourceAccessor.java

```diff
@@ -0,0 +1,987 @@
+package ikrs.httpd.resource;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.text.ParseException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.MissingResourceException;
+import java.util.UUID;
+import java.util.logging.Level;
+
+import ikrs.httpd.AuthorizationException;
+import ikrs.httpd.ConfigurationException;
+import ikrs.httpd.Constants;
+import ikrs.httpd.CustomUtil;
+import ikrs.httpd.DataFormatException;
+import ikrs.httpd.FileHandler;
+import ikrs.httpd.HeaderFormatException;
+import ikrs.httpd.HTTPConnectionUserID;
+import ikrs.httpd.HTTPHandler;
+import ikrs.httpd.HTTPHeaders;
+import ikrs.httpd.MD5;
+import ikrs.httpd.MissingParamException;
+import ikrs.httpd.PostDataWrapper;
+import ikrs.httpd.Resource;
+import ikrs.httpd.ResourceAccessor;
+import ikrs.httpd.UnsupportedFormatException;
+
+
+import ikrs.typesystem.BasicNumberType;
+import ikrs.typesystem.BasicStringType;
+import ikrs.typesystem.BasicType;
+import ikrs.util.CustomLogger;
+import ikrs.util.DefaultEnvironment;
+import ikrs.util.Environment;
+import ikrs.util.MIMEType;
+import ikrs.util.session.Session;
+
+import ikrs.io.fileio.htaccess.HypertextAccessFile;
+
+/**
+ * The FileSystemResourceAccessor is the default implementation of ResourceAccessor that handles
+ * requests to the document root respective to the file system (the document root is part of the
+ * file system).
+ *
+ * @autor Ikaros Kappler
+ * @date 2012-07-23
+ * @version 1.0.0
+ **/
+
+
+public class FileSystemResourceAccessor
+    extends AbstractResourceAccessor 
+    implements ResourceAccessor {
+
+
+    private HypertextAccessHandler hypertextAccessHandler;
+    
+    private CustomLogger logger;
+
+    /**
+     * ... 
+     **/
+    public FileSystemResourceAccessor( HTTPHandler handler,
+				       CustomLogger logger ) {
+	super( handler );
+	
+	
+	this.hypertextAccessHandler = new HypertextAccessHandler( logger, 
+								  false    // strictMode
+								  );
+	this.logger = logger;
+    }
+
+    private CustomLogger getLogger() {
+	return this.logger;
+    }
+
+    /**
+     * This method locates the desired resource addressed by the given URI.
+     *
+     * @throws ResouceMissingException If the specified resource cannot be found.
+     * @throws AuthorizationException If the requested resource requires authorization.
+     * @throws HeaderFormatException If the passed headers are malformed.
+     * @throws DataFormatException If the passed data is malformed.
+     * @throws ConfigurationException If the system encounters any server configuration issues.
+     * @throws SecurityException If the requested resource is not accessible (forbidden).
+     * @throws IOException If any IO errors occur.
+     **/
+    public Resource locate( URI uri,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 22. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 231-385 into a separate method.